### PR TITLE
Update monitoring

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

**Review by mlohvynenko — APPROVED** [review]

File: database/database.go [hunk 1]
```diff
@@ -364,31 +362,31 @@ func (db *Database) RemoveInstance(instanceID string) (err error) {
...
+) (ident aostypes.InstanceIdent, version string, err error) {
 	instance, err := db.getInstanceInfoFromQuery("SELECT * FROM instances WHERE instanceID = ?", instanceID)
 	if err != nil {
-		return ident, 0, aoserrors.Wrap(err)
+		return ident, "", aoserrors.Wrap(err)
 	}
 
 	stmt, err := db.sql.Prepare(
-		`SELECT aosVersion FROM services WHERE aosVersion = (SELECT MAX(aosVersion)
+		`SELECT version FROM services WHERE version = (SELECT MAX(version)
```
Comment (mykola-kobets-epam) [inline database/database.go:372]: max shouldn't work with version

File: servicemanager/servicemanager.go [hunk 2]
```diff
@@ -194,6 +196,20 @@ func (sm *ServiceManager) GetServiceInfo(serviceID string) (serviceInfo ServiceI
 		return serviceInfo, aoserrors.Wrap(err)
 	}
 
+	sort.Slice(services, func(i, j int) bool {
+		ver1, err := semver.NewSemver(services[i].Version)
+		if err != nil {
+			return services[i].Version < services[j].Version
```
Comment (mykola-kobets-epam) [inline servicemanager/servicemanager.go:202]: semver should be used

File: iamclient/iamclient.go [hunk 3]
```diff
@@ -256,23 +252,68 @@ func (client *Client) Close() (err error) {
...
 	}
 
 	log.WithFields(log.Fields{
 		"nodeID":   response.GetNodeId(),
 		"nodeType": response.GetNodeType(),
 	}).Debug("Get node Info")
 
-	return response.GetNodeId(), response.GetNodeType(), nil
-}
+	client.nodeInfo = cloudprotocol.NodeInfo{
```
Comment (mykola-kobets-epam) [inline iamclient/iamclient.go:269]: there were some convertion functions in aos_common, can be utilized here I think.

**Review by mykola-kobets-epam — APPROVED** [review]

File: resourcemanager/resourcemanager.go [hunk 4]
```diff
@@ -111,12 +111,12 @@ func New(nodeType, unitConfigFile string, alertSender AlertSender) (resourcemana
 	return resourcemanager, nil
 }
 
-// GetUnitConfigInfo returns unit config info.
-func (resourcemanager *ResourceManager) GetUnitConfigInfo() (version string) {
+// GetUnitConfigStatus returns unit config status.
+func (resourcemanager *ResourceManager) GetUnitConfigStatus() (version string, err error) {
```
[the remark this answers is not in the document]
Comment (al1img) [reply resourcemanager/resourcemanager.go:116]: It is return status indeed: err indicates if something wrong with unit config. Nil error means status ok, not nil - status error.

File: database/database.go [hunk 5]
```diff
@@ -50,7 +50,7 @@ const (
 	syncMode    = "NORMAL"
 )
 
-const dbVersion = 6
+const dbVersion = 7
```
Comment (MykolaSolyanko) [inline database/database.go:55]: i think we should update migration script

**Review by MykolaSolyanko — APPROVED** [review]

File: database/database.go (hunk 5 above)
Comment (al1img) [reply database/database.go:55]: We will not perform migration for this release.

**Pull request merged**